model/riscv_insts_vext_mask.sail

```diff
@@ -38,24 +38,24 @@ function clause execute(MMTYPE(funct6, vs2, vs1, vd)) = {
   let 'n = num_elem;
   let 'm = SEW;
 
-  let vs1_val : vector('n, bool) = read_vmask(num_elem, 0b0, vs1);
-  let vs2_val : vector('n, bool) = read_vmask(num_elem, 0b0, vs2);
-  let vd_val  : vector('n, bool) = read_vmask(num_elem, 0b0, vd);
+  let vs1_val : bits('n) = read_vmask(num_elem, 0b0, vs1);
+  let vs2_val : bits('n) = read_vmask(num_elem, 0b0, vs2);
+  let vd_val  : bits('n) = read_vmask(num_elem, 0b0, vd);
 
   let (initial_result, mask) = init_masked_result_carry(num_elem, SEW, 0, vd_val);
   var result = initial_result;
 
   foreach (i from 0 to (num_elem - 1)) {
-    if mask[i] then {
+    if mask[i] == bitone then {
       result[i] = match funct6 {
-        MM_VMAND     => vs2_val[i] & vs1_val[i],
-        MM_VMNAND    => not(vs2_val[i] & vs1_val[i]),
-        MM_VMANDN    => vs2_val[i] & not(vs1_val[i]),
-        MM_VMXOR     => vs2_val[i] != vs1_val[i],
-        MM_VMOR      => vs2_val[i] | vs1_val[i],
-        MM_VMNOR     => not(vs2_val[i] | vs1_val[i]),
-        MM_VMORN     => vs2_val[i] | not(vs1_val[i]),
-        MM_VMXNOR    => vs2_val[i] == vs1_val[i]
+        MM_VMAND     => (vs2_val & vs1_val)[i],
+        MM_VMNAND    => ~(vs2_val & vs1_val)[i],
+        MM_VMANDN    => (vs2_val & (vs1_val ^ ones()))[i],
```

Comment:
Use ~ here?

Reply:
This whole block could be simplified by hoisting the full bits operation outside of the foreach. But that cleanup can be done later.
I'd suggest using ~ here instead of cor with ones though.